### DISCUSSION TITLE
Bug - field array fields removed when useFieldArray hook is used in parent and child component

### DISCUSSION THREAD

**Discussion by wdfinch** [react-hook-form/react-hook-form#5969] [Q&A]
**Describe the bug**

**if** 
1. I have a parent component and child component that both call `useFieldArray` with the same-named field array
2. I mount the child and add some elements to the field array
3. I unmount the child field array and then remount it

**then**
The `fields` property of the field array will be empty rather than containing the values in `formValues`. 

**To Reproduce**

1. Go to the sandbox
2. Click add element and type some input
3. Click the toggle show child button twice to unmount and remount the child component
4. Notice that fields is now empty while `formValues` still has the field array values

**Codesandbox link (Required)**
https://codesandbox.io/s/nice-monad-rkydl?file=/src/App.js

**Expected behavior**
I expect that fields would have the `formValues` for the field array when it's remounted.

**Additional context**
The only reason I am mounting `useFieldArray` twice in the parent and child component is because in my app I need to use `setValue` to modify the field array values before the field array is first mounted. If I don't call `useFieldArray`, then I can't use `setValue` on field array elements. If this is no longer the case, then I would gladly remove the `useFieldArray` call in the parent.

Additionally, @bluebill1049 mentioned that using calling `useFieldArray` twice was safe when I last asked. Additionally, this did work in older versions which leads me to believe it is just a bug.

**Comment by bluebill1049** [comment]
<img width="435" alt="Screen Shot 2021-07-16 at 4 01 29 pm" src="https://user-images.githubusercontent.com/10513364/125899754-e5b9d92d-dd57-4103-bd8e-2855e7df34f5.png">

i almost miss that, why there is another useFieldArray?

**Reply by wdfinch** [reply]
So in past versions, certain operations on field arrays did not work unless the `useFieldArray` hook was called. So for example, if I wanted to use `setValue` to change a field array before the field array was mounted, it wouldn't work. The solution was to call the hook for the field array before running the modify function like `setValue`.

This is why I was calling it. As mentioned above if this no longer the case, then I can remove it. I just tested and it seems like `setValue` is working without calling the hook so maybe this was changed. 

I guess another use case is the user would like to use field array methods in multiple places. You mentioned the new  `update` methods should work without the field array being mounted in the RFC. Maybe we should make a note in the docs that multiple calls to useFieldArray results in undefined behavior if this is the case.

**Reply by wdfinch** [reply]
My bug was just based on how things used to work and that they seem to be working differently now. I'm never quite sure what will happen between version with field arrays when you don't have every element mounted.

**Comment by wdfinch** [comment]
After some testing, it seems things work now in my app without having to mount the field array twice. Just didn't know this would work now as I didn't see anything in the change log about this. Sorry and didn't mean to waste your time.

**Comment by wdfinch** [comment]
It appears that omitting the root `useFieldArray` results is setValue follow by append causing a crash. See #6010.
